language/transaction-builder-generator/src/common.rs

```diff
@@ -13,12 +17,58 @@ pub(crate) fn type_not_allowed(type_tag: &TypeTag) -> ! {
 
 /// Clean up doc comments extracter by the Move prover.
 pub(crate) fn prepare_doc_string(doc: &str) -> String {
-    let mut doc = doc.replace("\n", " ").trim().to_string();
-    loop {
-        let doc2 = doc.replace("  ", " ");
-        if doc == doc2 {
-            return doc;
+    doc.replace("\n ", "\n").trim().to_string()
+}
+
+fn quote_type_as_format(type_tag: &TypeTag) -> Format {
+    use TypeTag::*;
+    match type_tag {
+        Bool => Format::Bool,
+        U8 => Format::U8,
+        U64 => Format::U64,
+        U128 => Format::U128,
+        Address => Format::TypeName("AccountAddress".into()),
+        Vector(type_tag) => match type_tag.as_ref() {
+            U8 => Format::Bytes,
+            _ => type_not_allowed(type_tag),
```

Comment:
We may have to extend this to handle the upcoming changes to `TransactionArgument` (but a TODO placeholder seems fine for now).

Reply:
https://github.com/libra/libra/pull/5350 will make sure that we catch missing cases (at least in `rust.rs`) before landing a script that uses them.